frontend/server/src/main/java/com/amazonaws/ml/mms/http/ManagementRequestHandler.java

```diff
@@ -239,11 +243,20 @@ private void handleRegisterModel(
 
     private void handleUnregisterModel(ChannelHandlerContext ctx, String modelName)
             throws ModelNotFoundException {
+        logger.warn("Unregister request");
         ModelManager modelManager = ModelManager.getInstance();
-        if (!modelManager.unregisterModel(modelName)) {
+        // TODO AW-10/28/19: Will need to have this return an enum so we can track multiple errors.
+        // Or just return the raw httpstatus so we can bubble up
+        HttpResponseStatus httpResponseStatus = modelManager.unregisterModel(modelName);
```

Comment:
I think unregisterModel sending a HttpResponseStatus is a good change. Yes, you could probably look into simplifying the if/else conditions by either bubbling it up or just invoking NettyUtils.sendError() here itself.